### ISSUE TITLE
errors concerning gpu_... when moving to v2.6.2.0-dev

### ISSUE THREAD

**Issue by kfjahnke** [AcademySoftwareFoundation/OpenImageIO#4389]
Compiling master and v2.6.2.0-dev on my debian box, I get this error:
```
[ 37%] Building CXX object src/libOpenImageIO/CMakeFiles/OpenImageIO.dir/imageioplugin.cpp.o
/home/kfj/src/OpenImageIO/src/libOpenImageIO/imageio.cpp: In function ‘bool OpenImageIO_v2_6_2::attribute(string_view, TypeDesc, const void*)’:
/home/kfj/src/OpenImageIO/src/libOpenImageIO/imageio.cpp:391:21: error: ‘gpu_attribute’ is not a member of ‘OpenImageIO_v2_6_2::pvt’
  391 |         return pvt::gpu_attribute(name, type, val);
      |                     ^~~~~~~~~~~~~
/home/kfj/src/OpenImageIO/src/libOpenImageIO/imageio.cpp: In function ‘bool OpenImageIO_v2_6_2::getattribute(string_view, TypeDesc, void*)’:
/home/kfj/src/OpenImageIO/src/libOpenImageIO/imageio.cpp:496:21: error: ‘gpu_getattribute’ is not a member of ‘OpenImageIO_v2_6_2::pvt’
  496 |         return pvt::gpu_getattribute(name, type, val);
      |                     ^~~~~~~~~~~~~~~~

```
Did a regression to v2.6.1.1-dev, where the build runs through just fine.

**Comment by lgritz** [comment]
I don't understand how that's possible. 

In imageio_pvt.h, the declaration is very clearly in the pvt namespace.

Is it possible that you have multiple OIIO's installed, or something out of place, and somehow you are encountering an older imageio_pvt.h from before these functions were added?

**Comment by kfjahnke** [comment]
> Is it possible that you have multiple OIIO's installed, or something out of place

Thanks for the hint. I did a complete uninstall, cloned the repo anew and did a build from master, which ran through without errors. Sorry for the noise!

**Comment by lgritz** [comment]
At some point not too long ago, I believe imageio_pvt.h (which is not a public header) might have changed location internally within the project. So it's possible that if you updated the source tree on top of an old one (versus a totally clean checkout in a new place), maybe you had two versions of that header and it was picking up the old one?

**Comment by kfjahnke** [comment]
On 26.08.24 23:53, Larry Gritz wrote:

> At some point not too long ago, I believe imageio_pvt.h (which is not a 
> public header) might have changed location internally within the 
> project. So it's possible that if you updated the source tree on top of 
> an old one (versus a totally clean checkout in a new place), maybe you 
> had two versions of that header and it was picking up the old one?

Sounds plausible. I had the cloned repo here for a good while and was 
messing about with it - the project is really large and things like that 
can happen when you're trying to wrestle it down to your needs ;-)

Never mind, I shouldn't have raised an issue straight away but done 
instead what I did after your hint. Sorry again for the noise.

All is well now, lux runs with OIIO master. I did some poking about in 
OIIO because my colleague with apple silicon is going to come round to 
build a new lux on apple silicon binary, and he in turn had issues with 
the build (tripped up on the ptex version change) - the macPorts version 
is really old, so we're building OIIO from source. Maybe we should 
switch to homebrew which is offering a much more recent version...
